core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -110,6 +110,21 @@ private[spark] class Executor(
   // finish downloading dependencies.
   private val updateDependenciesLock = new ReentrantLock()
 
+  private lazy val mdcIsSupported = {
+    try {
+      // This tests if any class initialization error is thrown
+      val testKey = System.nanoTime().toString
+      MDC.put(testKey, "testValue")
+      MDC.remove(testKey)
+
+      true
+    } catch {
+      case t: Throwable =>
+        logInfo("MDC is not supported.", t)
+        false
+    }
+  }
+
```

Comment:
QQ: Can this not be moved to the companion object ?

Reply:
Good idea. Updated.